controllers/infinispan_controller.go

```diff
@@ -290,6 +290,37 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 		}
 	}
 
+	if infinispan.HasSites() {
+		reqLogger.Info("Creating the Cross-Site Deployment (Gossip Router)")
+		tunnel := r.GetGossipRouterDeployment(infinispan)
+		_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, tunnel, func() error {
+			return controllerutil.SetControllerReference(r.infinispan, tunnel, r.scheme)
```

Comment:
You just update controller reference in the mutate function, not required Deployment parts.
Please have a look how others controllers (service/config, etc.) performing updates for mutating fields

Reply:
I have no idea what to do TBH. 
The service seems to copy every field manually (which seems error-prone to me). Configs, secrets, batch just invoke `SetControllerReference()`.

Reply:
The code should like this
```
		tunnelDeployment := &appsv1.Deployment{
			ObjectMeta: metav1.ObjectMeta{
				Name:      infinispan.Name,
				Namespace: infinispan.Namespace,
			},
		}
		result, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, tunnelDeployment, func() error {
			tunnel := r.GetGossipRouterDeployment(infinispan)
			tunnelDeployment.Spec = tunnel.Spec
			tunnelDeployment.Labels = tunnel.Labels
			if tunnelDeployment.CreationTimestamp.IsZero() {
				reqLogger.Info("Creating the Cross-Site Deployment (Gossip Router)")
				return controllerutil.SetControllerReference(r.infinispan, tunnelDeployment, r.scheme)
			}
			return nil
		})
		if err != nil {
			if errors.IsConflict(err) {
				return reconcile.Result{Requeue: true}, nil
			} else {
				reqLogger.Error(err, "Failed to configure Cross-Site Deployment")
				return reconcile.Result{}, err
			}
		}
		if result == controllerutil.OperationResultCreated {
			reqLogger.Info("Cross-Site Deployment created")
		}
```